02/app.js

```diff
@@ -0,0 +1,39 @@
+const formEl = document.querySelector('form');
+formEl.noValidate = true;
+var regExEmailInput = /^([a-z0-9]){1}([\w-])*([a-z0-9]){1}([@]){1}([a-z0-9]){1}([\w\-\.])*(\.)([a-z0-9]){1}([a-z0-9]){1}$/i;
+
+formEl.addEventListener('submit', validatation);
+
+function validatation (e) {
+    e.preventDefault();
+    const emailValue = formEl.login.value;
+    const emailCheck = regExEmailInput.test(emailValue);
+    const password1Value = formEl.pass1.value;
+    const password2Value = formEl.pass2.value;
+    const checkboxValue = formEl.accept.checked;
+    const errors = [];
+
+    if(!emailCheck) {
+        errors.push(formEl.login);
+    }
+
+    if(password1Value.length < 6 || password2Value.length < 6 ) {
+        errors.push(formEl.pass1);
+    }
+
+    if(password1Value !== password2Value) {
+        errors.push(formEl.pass1, formEl.pass2);
+    }
+
+    if(!checkboxValue) {
+        errors.push(formEl.accept);
+    }
+    
+    if(errors.length === 0) {
+        console.log('DONE');
+    } else {
+        errors.forEach(function(el) {
+            el.previousElementSibling.style.color = 'red';
+        });
+    }
```

Comment:
Jeśli część danych jest niepoprawna, a potem je poprawisz to i tak są one oznaczone na czerwono chociaż są prawidłowo. Spróbuj to poprawić ;)